src/render_tree/text.rs

```diff
@@ -0,0 +1,80 @@
+static DEFAULT_FONT: &str = "Times New Roman";
+
+#[derive(Debug, PartialEq)]
+pub struct TextNode {
+    /// Body of the text node that will be drawn
+    pub value: String,
+    // TODO: change this to a stack-allocated string
+    pub font: String,
+    pub font_size: f64,
+    pub is_bold: bool,
+    // TODO: color, styles, visiblity, etc
+}
+
+impl TextNode {
+    /*
+        NOTE: I got the default font sizes from https://stackoverflow.com/a/70720104
+    */
+    pub fn new_heading1() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 37.,
+            is_bold: true,
+        }
+    }
+
+    pub fn new_heading2() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 27.5,
+            is_bold: true,
+        }
+    }
+
+    pub fn new_heading3() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 21.5,
+            is_bold: true,
+        }
+    }
+
+    pub fn new_heading4() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 18.5,
+            is_bold: true,
+        }
+    }
+
+    pub fn new_heading5() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 15.5,
+            is_bold: true,
+        }
+    }
+
+    pub fn new_heading6() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 12.,
+            is_bold: true,
+        }
+    }
+
+    pub fn new_paragraph() -> Self {
+        Self {
+            value: "".to_owned(),
+            font: DEFAULT_FONT.to_owned(),
+            font_size: 18.5,
+            is_bold: false,
+        }
+    }
```

Comment:
same for here, I would think about a function like `new_text` and `new_headingX` calls into it and passes the font_size to it.

Reply:
I meant, you define the fontsize in the function, we can have a functions `new_headingX` and `new_paragraph` that calls into a (private) function `new_text`, which wants the font size (this is defined in the public function. It would look like this then:


```rust
pub fn new_paragraph() -> {
   Self::new_text(18.5)
}
```

If you want, i can commit this as a extra commit and if we only want the others, we can just rebase without it.

Reply:
Ohhh gotcha, ok I misunderstood. In that case it would probably be better

Reply:
I have to start work soon so I will be able to review around lunch time, approx 12pm EDT